src/esp32/motor.rs

```diff
@@ -77,6 +78,7 @@ pub(crate) fn register_models(registry: &mut ComponentRegistry) {
 pub struct EncodedMotor<M, Enc> {
     motor: M,
     enc: Enc,
+    max_rpm: f64,
```

Comment:
I don't think we should bother storing `max_rpm` here or in `SingleEncodedMotor`. These motors should just implement GoFor by calling it on the underlying motor

Reply:
removed the fields from encoded motor and defer to internal moto's go_for until we add the more advanced encoder logic